app/services/notifications_cleanup_service.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class NotificationsCleanupService
+  # This can be expensive, so instead of cleaning up everything
+  # at once, we limit the number of accounts per run and run
+  # this more often.
+  ACCOUNTS_PER_RUN = 5
```

Comment:
I chose a very low number here to start out with. A larger number might make this run much too long on larger instances that have been around for some time. I think it can be raised to a much higher value once the bulk of years-old notifications is gone.